src/user-management/script.js

```diff
@@ -0,0 +1,127 @@
+  addClass(listContainerElement, 'tile-view');
+  listContainerElement.childNodes.forEach((listElement) => {
+    const imgElement = listElement.firstChild;
+    addClass(imgElement, 'element-display-remove');
+    addClass(listElement, 'tile-width');
+  });
+}
+
+function showTableView() {
+  tableViewActive = true;
+  tileViewActive = false;
+  tileViewBtn.classList.remove('btn-active');
+  tableViewBtn.classList.add('btn-active');
+  const listContainerElement = userListElement.lastChild;
+  removeClass(listContainerElement, 'tile-view');
+  addClass(listContainerElement, 'table-view');
+  listContainerElement.childNodes.forEach((listElement) => {
+    const imgElement = listElement.firstChild;
+    removeClass(imgElement, 'element-display-remove');
+    removeClass(listElement, 'tile-width');
+  });
+}
+
+function debounce(func, delay) {
+  let timerId;
+  return (...args) => {
+    if (timerId) {
+      clearTimeout(timerId);
+    }
+
+    timerId = setTimeout(() => {
+      func(...args);
+    }, delay);
+  };
+}
+
+userSearchElement.addEventListener(
+  'input',
+  debounce((event) => filterUsers(event.target.value), 500),
+);
+
+let usersDataList = [];
+
+async function getUsersData() {
+  try {
+    const usersRequest = await makeApiCall(RDS_API_USERS);
+    if (usersRequest.status === 200) {
+      usersDataList = await usersRequest.json();
+      usersDataList = usersDataList.members;
+      usersDataList = usersDataList.filter((user) => user.first_name);
+      usersDataList = usersDataList.map((user) => ({
+        first_name: user.first_name,
+        last_name: user.last_name ? user.last_name : '',
+        picture: user.picture && user.picture.url ? user.picture.url : '',
+      }));
+    }
+    generateUserList(usersDataList);
+  } catch (err) {
+    const paraELe = document.createElement('p');
+    const textNode = document.createTextNode('Something Went Wrong');
+    paraELe.appendChild(textNode);
+    addClass(paraELe, 'error-text');
+    addClass(loaderElement, 'element-display-remove');
+    userListElement.appendChild(paraELe);
+  }
+}
+
+async function generateUserList(users) {
+  userListElement.innerHTML = '';
+  if (!users.length) {
+    const paraELe = document.createElement('p');
+    const textNode = document.createTextNode('No data found');
+    paraELe.appendChild(textNode);
+    addClass(paraELe, 'error-text');
+    userListElement.appendChild(paraELe);
+    return;
+  }
+  const ulElement = document.createElement('ul');
+  users.forEach((userData) => {
+    const listElement = document.createElement('li');
+    const imgElement = document.createElement('img');
+    imgElement.src = userData.picture ? userData.picture : defaultAvatar;
+    imgElement.setAttribute('width', 70);
+    imgElement.setAttribute('height', 70);
+    const pElement = document.createElement('p');
+    let name =
+      userData.first_name.slice(0, 15) + ' ' + userData.last_name.slice(0, 15);
```

Comment:
Why is the name being sliced?

Reply:
So the reason I am slicing here is, on users API on staging there was not so many users information, so I tested this feature with members API too, and there I found this user name
<img width="536" alt="image" src="https://user-images.githubusercontent.com/54801934/200492203-63a6af39-38c3-42e9-982e-6fed93203553.png">
though this is not accurate data, in case if get such incorrect data so I sliced the name. 
But the slicing name is wrong, will remove it. I think we should have validation on the name when we are taking user information if in case we are not having it right now.

Reply:
Good observation. Instead of being declarative, we can let the last layer handle it. Read this https://developer.mozilla.org/en-US/docs/Web/CSS/text-overflow.